src/shogun/regression/LeastAngleRegression.h

```diff
@@ -124,10 +124,9 @@ class CLeastAngleRegression: public CLinearMachine
 	
 	void switch_w(int32_t num_variable)
 	{
-		if (w.vlen <= 0)
-			SG_ERROR("Please train the model before updating its parameters")
-		if (size_t(num_variable) >= m_beta_idx.size() || num_variable < 0)
-			SG_ERROR("cannot switch to an estimator of %d non-zero coefficients", num_variable)
+		REQUIRE(w.vlen > 0,"Please train the model before updating its parameters")
```

Comment:
newline missing.
Also, is this a message for the user? Then it needs to be more concrete, i.e. how to solve it?